src/client/App.js

```diff
@@ -1,33 +1,53 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { ConnectedRouter } from 'connected-react-router';
-import { Switch, Route } from 'react-router-dom';
-import PrivateRoute from './components/PrivateRoute/PrivateRoute';
-import GuestRoute from './components/GuestRoute/GuestRoute';
-import CheckIfLoggedIn from './components/CheckIfLoggedIn/CheckIfLoggedIn';
+import React, { useEffect, useState } from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import { Login, Registro, Home } from './pages/';
+import usuarioContexto from './state/state';
 
-import LoginPage from './pages/Login/Login';
-import RegisterPage from './pages/Register/Register';
-import DashboardPage from './pages/Dashboard/Dashboard';
-import PageNotFound from './pages/NotFound/NotFound';
+export default function App() {
+  const [user, setUser] = useState({ nombre: '', uid: '', islogged: false });
 
-const App = props => {
-  return (
-    <ConnectedRouter history={props.history}>
-      <CheckIfLoggedIn>
-        <Switch>
-          <PrivateRoute exact path="/" component={DashboardPage} />
-          <GuestRoute exact path="/login" component={LoginPage} />
-          <GuestRoute exact path="/register" component={RegisterPage} />
-          <Route component={PageNotFound} />
-        </Switch>
-      </CheckIfLoggedIn>
-    </ConnectedRouter>
-  );
-};
+  useEffect(() => {
+    // aqui pasan cosas  muy raras piter
+    const token = localStorage.getItem('token') || null;
+    const existToken = Boolean(token);
 
-App.propTypes = {
-  history: PropTypes.object
-};
+    if (existToken) {
+      fetch('http://localhost:3000/api/renew', {
+        headers: {
+          token
+        }
+      })
+        .then(res => res.json())
+        .then(data => {
+          if (data.ok) {
+            setUser(p => ({
+              ...p,
+              nombre: data.usuario.nombre,
+              uid: data.usuario.uid,
+              islogged: true
+            }));
+          }
+        });
+    }
+  }, []);
 
-export default App;
+  return (
+    <usuarioContexto.Provider value={{ user, setUser }}>
```

Comment:
You are exposing a function to directly mutate the user state. I would rather create function `login` and `logout` that mutates the state. In those functions you could add some validation to check that the given object is indeed a valid user for instance and not a random object.